bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/ThingManagerImpl.java

```diff
@@ -1249,6 +1249,18 @@ public boolean isEnabled(ThingUID thingUID) {
         return !isDisabledByStorage(thingUID);
     }
 
+    @Override
+    public @Nullable ThingStatusInfo getStatusInfo(ThingUID thingUID) {
+        Thing thing = getThing(thingUID);
```

Comment:
Just as a reminder for myself or future improvement: The `ThingManagerImpl` stores things in a `Set` and iterates the elements to get the thing with given `ThingUID` in the `getThing()` method. We maybe want to replace the `Set` by a `Map`.